File: src/test/unit/languages/python/fixtures/imports.py
# Import patterns and module dependencies

# Standard library imports
from typing import Optional, List, Dict
import json

# Relative imports (simulated - would come from other files)
# from .class_based import UserService, UserRepository
# from .type_hints import User, UserDTO
# from .functional import validate_user, create_user


class ImportingService:
    """Service that imports and uses other classes"""
    
    def __init__(self):
        # In real code, would use imported classes
        self.service = None  # Would be: UserService(UserRepository())
        self.users: List = []  # Would be: List[User]
    
    def process(self, data: Dict) -> Optional:
        """Process with type hints"""
        # Would call: validate_user(data)
        # Would call: create_user(data)
        return None


def transform_data(raw: dict) -> dict:
    """Transform raw data"""
    return json.loads(json.dumps(raw))


class DataProcessor:
    """Processor using imported functions"""
    
    def process_item(self, item: dict):
        """Process single item"""
        # Would call imported function: validate_user(item)
        return transform_data(item)

File: packages/analysis/src/test/unit/languages/javascript/fixtures/imports.js
// Import patterns (CommonJS and ES6 style)

// CommonJS imports
const path = require('path');
const fs = require('fs');

// Simulated imports from other modules
// const { UserService, UserRepository } = require('./class-based');
// const { validateUser, createUser } = require('./functional');

class ImportingService {
  constructor() {
    this.service = null; // Would be: new UserService(new UserRepository())
    this.users = [];
  }

  process(data) {
    // Would call: validateUser(data)
    // Would call: createUser(data)
    return null;
  }
}

function transformData(raw) {
  return JSON.parse(JSON.stringify(raw));
}

class DataProcessor {
  processItem(item) {
    // Would call imported function: validateUser(item)
    return transformData(item);
  }
}

module.exports = { ImportingService, DataProcessor, transformData };
